File: VB/DateAndTimeColumns_Properties/Item.vb
Imports DevExpress.Mvvm

Namespace DateAndTimeColumns_Properties

    Public Class Item
        Inherits BindableBase

        Private dateTimeField As Date

        Public Property DateTime As Date
            Get
                Return dateTimeField
            End Get

            Set(ByVal value As Date)
                dateTimeField = value
                RaisePropertiesChanged()
            End Set
        End Property

        Public Property [Date] As Date
            Get
                Return dateTimeField.Date
            End Get

            Set(ByVal value As Date)
                Dim time = dateTimeField.TimeOfDay
                DateTime = value.Date + time
                RaisePropertiesChanged()
            End Set
        End Property

        Public Property Time As Date
            Get
                Return Date.Today.AddTicks(dateTimeField.TimeOfDay.Ticks)
            End Get

            Set(ByVal value As Date)
                DateTime = dateTimeField.Date.AddTicks(value.TimeOfDay.Ticks)
                RaisePropertiesChanged()
            End Set
        End Property
    End Class
End Namespace

File: CS/DateAndTimeColumns_Properties/Item.cs
using DevExpress.Mvvm;
using System;

namespace DateAndTimeColumns_Properties
{
    public class Item : BindableBase
    {
        private DateTime dateTime;
        public DateTime DateTime {
            get => dateTime;
            set {
                dateTime = value;
                RaisePropertiesChanged();
            }
        }

        public DateTime Date {
            get => dateTime.Date;
            set {
                var time = dateTime.TimeOfDay;
                DateTime = value.Date + time;
                RaisePropertiesChanged();
            }
        }

        public DateTime Time {
            get => DateTime.Today.AddTicks(dateTime.TimeOfDay.Ticks);
            set {
                DateTime = dateTime.Date.AddTicks(value.TimeOfDay.Ticks);
                RaisePropertiesChanged();
            }
        }
    }
}
